appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1181,11 +1186,22 @@ private void scanArgs(String[] args) {
                         }
                         break;
                     case deploy:
-                        File deployment = new File(value);
+                        // check for context root definition
                         if (deployments == null) {
                             deployments = new LinkedList<>();
+                            contextRoots = new Properties();
+                        }                        
+                        String fileName = value;
+                        String deployContext = null;
+                        if (value.contains(":")) {
```

Comment:
Would this work on Windows with fully qualified paths?
`--deploy C:\Users\AndrewPielage\Downloads\blah.war:/myApp`

Reply:
Having looked into it a bit since I should do a short video on the usage for marketing I would say if it is valid to use a qualified path like `C:\...` the `value.contains(":")` check is not sufficient and the splitting of the root-context below also has to be adopted since `indexOf` could wrongly point to the `C:`.

Reply:
OK on windows maybe we need to use the ; as separator

Reply:
Now uses File.pathSeparator to split